.github/actions/test-gradle-project/action.yaml

```diff
@@ -4,15 +4,20 @@ inputs:
   project-name:
     description: "Name of gradle project to test"
     required: true
+  java-version:
+    description: "JDK to use"
+    required: true
 
 runs:
   using: "composite"
   steps:
     - uses: ./.github/actions/setup-jdk
+      with:
+        java-version: ${{inputs.java-version}}
     - uses: ./.github/actions/setup-gradle-cache
     - name: Run tests
       shell: bash
-      run: ./gradlew ${{inputs.project-name}}:check --parallel
+      run: ./gradlew ${{inputs.project-name}}:test --parallel
```

Comment:
Is check not needed because we run the other things e.g spotless separately?

Reply:
Yes. Because spotless didn't work in [f98d019](https://github.com/neo4j/apoc/pull/532/commits/f98d0198d5615d21652e6a7980cbf5461abc8033) for Java 21 🤷